### PR TITLE
glew: test failure

### PR DESCRIPTION
**Pull request by rwhogg**

### CODE REVIEW

**Comment by rwhogg** [comment]
From https://travis-ci.org/Linuxbrew/homebrew-core/builds/157123801#L543-L571

```
/home/linuxbrew/.linuxbrew/bin/gcc-5
/tmp/glew-test-20160902-30072-17kguvj/test.c
-o
test
-L/home/linuxbrew/.linuxbrew/Cellar/glew/2.0.0/lib
-lGLEW
-framework
GLUT

gcc-5: error: GLUT: No such file or directory

gcc-5: error: unrecognized command line option '-framework'

Error: glew: failed

Failed executing: /home/linuxbrew/.linuxbrew/bin/gcc-5 /tmp/glew-test-20160902-30072-17kguvj/test.c -o test -L/home/linuxbrew/.linuxbrew/Cellar/glew/2.0.0/lib -lGLEW -framework GLUT

/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1584:in `block in system'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1521:in `open'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1521:in `system'
/home/linuxbrew/.linuxbrew/Library/Taps/homebrew/homebrew-core/Formula/glew.rb:46:in `block in <class:Glew>'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1425:in `block (2 levels) in run_test'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:762:in `with_logging'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1424:in `block in run_test'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:14:in `block in mktemp'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `block in run'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `chdir'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `run'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:13:in `mktemp'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1418:in `run_test'
/home/linuxbrew/.linuxbrew/Library/Homebrew/test.rb:33:in `block in <main>'
/home/travis/.rvm/rubies/ruby-2.0.0-p648-clang/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/home/linuxbrew/.linuxbrew/Library/Homebrew/test.rb:32:in `<main>'
```

Test failure, should be easily fixed by removing the reference to the macOS extension "-framework".

Frankly, I'm surprised this compiled at all, since I feel like it's missing a dependency on `linuxbrew/xorg/mesa` (cc @maxim-belkin, I'll want your approval before merging this). I suspect that it's incorrectly linking against host OpenGL libraries, but not sure.

**Comment by rwhogg** [comment]
From https://travis-ci.org/Linuxbrew/homebrew-core/builds/157214006#L543-L550

```
/home/linuxbrew/.linuxbrew/bin/gcc-5
/tmp/glew-test-20160902-30020-dpexjm/test.c
-o
test
-L/home/linuxbrew/.linuxbrew/Cellar/glew/2.0.0/lib
-lGLEW

/tmp/glew-test-20160902-30020-dpexjm/test.c:2:23: fatal error: GLUT/glut.h: No such file or directory
```

This is missing a dependency on `freeglut`, which exists in both `homebrew/x11` and `linuxbrew/xorg`. I'll prefer the latter.

**Comment by rwhogg** [comment]
`mesa` did not build.

```
checking for PTHREADSTUBS... no
configure: error: Package requirements (pthread-stubs) were not met:
No package 'pthread-stubs' found
Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.
```

@maxim-belkin, I'm aware that you've currently got https://github.com/Linuxbrew/homebrew-xorg/pull/168 underway, but are you okay with me making a minimal PR to add a dependency on `linuxbrew/xorg/libpthread-stubs` to `mesa` to allow this PR to move forward?

**Comment by maxim-belkin** [comment]
go ahead

File: Formula/glew.rb [hunk 1]
```diff
@@ -15,6 +15,8 @@ class Glew < Formula
 
   option :universal
 
+  depends_on "linuxbrew/xorg/freeglut" unless OS.mac?
```
Comment (rwhogg) [inline Formula/glew.rb]: @maxim-belkin should I be using the core `freeglut` here?

Comment (maxim-belkin) [inline Formula/glew.rb]: we removed freeglut from xorg when upstream added it to the corr

**Comment by maxim-belkin** [comment]
Attempting to build a bottle here: #2621

**Pull request closed without merging**